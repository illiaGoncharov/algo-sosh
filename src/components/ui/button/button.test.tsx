import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("button component tests", () => {
  // Снимок при наличии текста
  it("renders button with text", () => {
    const tree = renderer.create(<Button text="test" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Снимок компонента без текста
  it("renders button without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Снимок disabled кнопки
  it("renders disabled button correctly", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Снимок loading кнопки
  it("renders loading button correctly", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Тест взаимодействия
  it("calls the onClick callback when button is clicked", () => {
    // Подготовка: Устанавливаем кнопку с callback
    const callback = jest.fn();
    render(<Button onClick={callback} />);
    // Действие: Симулируем клик по кнопке
    const button = screen.getByRole("button");
    fireEvent.click(button);
    // Проверка: Убеждаемся, что обратный вызов был вызван
    expect(callback).toHaveBeenCalled();
  });
});
