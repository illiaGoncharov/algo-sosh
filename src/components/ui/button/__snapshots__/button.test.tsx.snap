// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button component tests renders button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`button component tests renders button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button component tests renders disabled button correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`button component tests renders loading button correctly 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
  />
</button>
`;
